src/java.base/share/classes/jdk/internal/reflect/SerializationBytecodeGenerator.java

```diff
@@ -0,0 +1,264 @@
+                    // stack: PutField fieldName <val>
+                    switch (fieldType.descriptorString()) {
+                        case "B" -> cb.invokevirtual(CD_ObjectOutputStream_PutField, "put", MTD_ObjectOutputStream_PutField_put_B);
+                        case "C" -> cb.invokevirtual(CD_ObjectOutputStream_PutField, "put", MTD_ObjectOutputStream_PutField_put_C);
+                        case "D" -> cb.invokevirtual(CD_ObjectOutputStream_PutField, "put", MTD_ObjectOutputStream_PutField_put_D);
+                        case "F" -> cb.invokevirtual(CD_ObjectOutputStream_PutField, "put", MTD_ObjectOutputStream_PutField_put_F);
+                        case "I" -> cb.invokevirtual(CD_ObjectOutputStream_PutField, "put", MTD_ObjectOutputStream_PutField_put_I);
+                        case "J" -> cb.invokevirtual(CD_ObjectOutputStream_PutField, "put", MTD_ObjectOutputStream_PutField_put_J);
+                        case "S" -> cb.invokevirtual(CD_ObjectOutputStream_PutField, "put", MTD_ObjectOutputStream_PutField_put_S);
+                        case "Z" -> cb.invokevirtual(CD_ObjectOutputStream_PutField, "put", MTD_ObjectOutputStream_PutField_put_Z);
+                        default -> cb.invokevirtual(CD_ObjectOutputStream_PutField, "put", MTD_ObjectOutputStream_PutField_put_L);
+                    }
+                    // stack: (empty)
+                }
+                // commit fields to stream
+                cb.aload(1);
+                cb.invokevirtual(CD_ObjectOutputStream, "writeFields", ConstantDescs.MTD_void);
+                cb.return_();
+            })
+        ));
+        try {
+            MethodHandles.Lookup hcLookup = lookup.defineHiddenClassWithClassData(bytes, List.copyOf(getters), true);
+            return hcLookup.findStatic(hcLookup.lookupClass(), "writeObject", MethodType.methodType(void.class, Object.class, ObjectOutputStream.class))
+                .asType(MethodType.methodType(void.class, cl, ObjectOutputStream.class));
+        } catch (IllegalAccessException | NoSuchMethodException e) {
+            throw new InternalError("Error in writeObject generation", e);
+        }
+    }
+
+    private static final ClassDesc CD_ObjectInputStream = ClassDesc.of("java.io.ObjectInputStream");
+    private static final ClassDesc CD_ObjectInputStream_GetField = ClassDesc.of("java.io.ObjectInputStream$GetField");
+
+    private static final ClassDesc CD_ObjectOutputStream = ClassDesc.of("java.io.ObjectOutputStream");
+    private static final ClassDesc CD_ObjectOutputStream_PutField = ClassDesc.of("java.io.ObjectOutputStream$PutField");
+
+    private static final ClassDesc CD_Generated_writeObject = ClassDesc.of("jdk.internal.reflect", "Generated$$writeObject");
+    private static final ClassDesc CD_Genearted_readObject = ClassDesc.of("jdk.internal.reflect", "Generated$$readObject");
```

Comment:
It might be better to use `ReferenceClassDescImpl​.ofValidated​(String)` here, or at least `ClassDesc​.ofDescriptor​(String)` to avoid binary to internal name conversion:

```suggestion
    private static final ClassDesc CD_ObjectInputStream = ReferenceClassDescImpl.ofValidated("Ljava/io/ObjectInputStream;");
    private static final ClassDesc CD_ObjectInputStream_GetField = ReferenceClassDescImpl.ofValidated("Ljava/io/ObjectInputStream$GetField;");

    private static final ClassDesc CD_ObjectOutputStream = ReferenceClassDescImpl.ofValidated("Ljava/io/ObjectOutputStream;");
    private static final ClassDesc CD_ObjectOutputStream_PutField = ReferenceClassDescImpl.ofValidated("Ljava/io/ObjectOutputStream$PutField;");

    private static final ClassDesc CD_Generated_writeObject = ReferenceClassDescImpl.ofValidated("Ljdk/internal/reflect/Generated$$writeObject;");
    private static final ClassDesc CD_Generated_readObject = ReferenceClassDescImpl.ofValidated("Ljdk/internal/reflect/Generated$$readObject;");
```

Reply:
Good idea.